pkg/monitor/cluster/hiveregistrationstatus.go

```diff
@@ -0,0 +1,60 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+type hiveRegistrationState struct {
+	isReachable        bool
+	notReachableReason string
+}
+
+func (mon *Monitor) emitHiveRegistrationStatus(ctx context.Context) error {
+	if mon.hiveClusterManager == nil {
+		return nil
+	}
+
+	state, err := mon.hiveRegistrationState(ctx)
+	if err != nil {
+		return err
+	}
+
+	if !state.isReachable {
+		mon.emitGauge("hive.registration.isUnreachable", 1, map[string]string{
+			"reason": state.notReachableReason,
+		})
+	}
+
+	return nil
```

Comment:
I still think that we should do `ClusterDeployment` conditions instead of this.

I realised that my [previous comments](https://github.com/Azure/ARO-RP/pull/2175#discussion_r915691248) we were missleading: I wanted to refer to conditions, not statuses. And a better example would be [`emitClusterVersionConditions`](https://github.com/Azure/ARO-RP/blob/3ed9676d3672283c4c94d07050da0d805da83634/pkg/monitor/cluster/clusterversionconditions.go) instead of deployment statuses. We have few other exampels such as `emitNodeConditions`, `emitPodConditions`, etc but they include more things which are irrelevant for cluster deployment statuses.

Cluster deployments have a lot of condiditions and we do not care about all of them.